packages/liveblocks-core/src/room.ts

```diff
@@ -795,6 +801,12 @@ export type PrivateRoomApi<M extends BaseMetadata> = {
   undoStack: readonly (readonly Readonly<HistoryOp<JsonObject>>[])[];
   nodeCount: number;
 
+  // For usage in Y.js provider
+  getProvider(): Provider | undefined;
+  setProvider(provider: Provider | undefined): void;
+
+  onProviderUpdate: Observable<void>;
```

Comment:
We now add three internal methods inside a `Room` object. 

1. `getProvider`: Returns the currently attached (Y.js) provider in a room.
2. `setProvider`: Attaches the specified provider in the room. Could be set to `undefined` (currently no usage).
3. `onProviderUpdate`: Subscription to changes in the provider attached to a room.